src/Data/List/ZipList.purs

```diff
@@ -25,6 +28,24 @@ newtype ZipList a = ZipList (List a)
 runZipList :: forall a. ZipList a -> List a
 runZipList (ZipList xs) = xs
 
+-- | Transpose any structure which is both `Foldable` and `Unfoldable`,
+-- | via `ZipList`; note that `sequence` for `ZipList` is the same as
+-- | `transpose`.
+transpose :: forall f a. (Foldable f, Unfoldable f) => f (f a) -> f (f a)
```

Comment:
We already have generalized type signatures for all kinds of things, though? `foldr`, `replicate`, `(<>)`, even `(+)` and `(*)`.

Just so that we're on the same page - are you suggesting we put specialized versions of `transpose` into `Data.List` and `Data.List.Lazy`, as well as this generalized version here?